src/LocalMapping.cc

```diff
@@ -730,16 +675,10 @@ void LocalMapping::SearchInNeighbors() {
 
     // Add some covisible of covisible
     // Extend to some second neighbors if abort is not requested
-    for (int i = 0, imax = vpTargetKFs.size(); i < imax; i++) {
-    const std::vector<KeyFrame*> vpSecondNeighKFs =
-            vpTargetKFs[i]->GetBestCovisibilityKeyFrames(20);
-    for (std::vector<KeyFrame*>::const_iterator vit2 = vpSecondNeighKFs.begin(),
-                                                vend2 = vpSecondNeighKFs.end();
-             vit2 != vend2; vit2++) {
-      KeyFrame* pKFi2 = *vit2;
-            if (pKFi2->isBad() ||
-                pKFi2->mnFuseTargetForKF == mpCurrentKeyFrame->mnId ||
-                pKFi2->mnId == mpCurrentKeyFrame->mnId)
+    for (KeyFrame *pKFi : vpTargetKFs) {
```

Comment:
This for loop modification needs to be reverted because inside the loop the list is appended to